packages/paypal-commerce-integration/src/paypal-commerce-credit/paypal-commerce-credit-button-strategy.ts

```diff
@@ -78,17 +84,30 @@ export default class PayPalCommerceCreditButtonStrategy implements CheckoutButto
             await this.paymentIntegrationService.loadDefaultCheckout();
         }
 
+        const state = this.paymentIntegrationService.getState();
+
         // Info: we are using provided currency code for buy now cart,
         // because checkout session is not available before buy now cart creation,
         // hence application will throw an error on getCartOrThrow method call
         const currencyCode = isBuyNowFlow
             ? providedCurrencyCode
-            : this.paymentIntegrationService.getState().getCartOrThrow().currency.code;
+            : state.getCartOrThrow().currency.code;
 
         await this.paypalCommerceIntegrationService.loadPayPalSdk(methodId, currencyCode, false);
 
         this.renderButton(containerId, methodId, paypalcommercecredit);
-        this.renderMessages(messagingContainerId);
+
+        if (currencyCode && messagingContainerId) {
```

Comment:
Lets add extra condition for `document.getElementById(messagingContainerId)` and remove it in `renderMessages` method